djangocms_video/models.py

```diff
@@ -15,104 +28,188 @@ def get_plugin_media_path(instance, filename):
         for information
         """
         return instance.get_media_path(filename)
-from cms.utils.compat.dj import python_2_unicode_compatible
 
-from . import settings
+
+# mp4, are required for full browser support
+ALLOWED_EXTENSIONS = getattr(
+    settings,
+    'DJANGOCMS_VIDEO_ALLOWED_EXTENSIONS',
+    ['mp4', 'webm', 'ogv'],
+)
+
+
+# Add additional choices through the ``settings.py``.
+def get_templates():
+    choices = getattr(
+        settings,
+        'DJANGOCMS_VIDEO_TEMPLATES',
+        [],
+    )
+    return choices
 
 
 @python_2_unicode_compatible
-class Video(CMSPlugin):
+class VideoPlayer(CMSPlugin):
+    """
+    Renders either an Iframe when ``link`` is provided or the HTML5 <video> tag
+    """
+    DEFAULT_CHOICE = 'default'
+
+    TEMPLATE_CHOICES = [
+        ('default', _('Default')),
+    ]
+
+    # The label will be displayed as help text in the structure board view.
+    template = models.CharField(
+        verbose_name=_('Template'),
+        choices=TEMPLATE_CHOICES + get_templates(),
+        default=DEFAULT_CHOICE,
+        max_length=50,
+    )
+    label = models.CharField(
+        verbose_name=_('Label'),
+        blank=True,
+        max_length=200,
```

Comment:
according to @czpython you don't have to. Is it okay to leave it out if its not required?